crates/accelerate/src/commutation_analysis.rs

```diff
@@ -0,0 +1,179 @@
+// that they have been altered from the originals.
+
+use pyo3::prelude::PyModule;
+use pyo3::{pyfunction, pymodule, wrap_pyfunction, Bound, PyResult, Python};
+use qiskit_circuit::operations::Param;
+use qiskit_circuit::Qubit;
+use smallvec::{smallvec, SmallVec};
+use std::hash::BuildHasherDefault;
+
+use crate::commutation_checker::CommutationChecker;
+use ahash::AHasher;
+use hashbrown::HashMap;
+use indexmap::IndexSet;
+use pyo3::prelude::*;
+
+use pyo3::types::{PyDict, PyList};
+use qiskit_circuit::dag_circuit::{DAGCircuit, NodeType, Wire};
+use rustworkx_core::petgraph::stable_graph::NodeIndex;
+
+type AIndexSet<T> = IndexSet<T, BuildHasherDefault<AHasher>>;
+#[derive(Clone, Debug)]
+pub enum CommutationSetEntry {
+    Index(usize),
+    SetExists(Vec<AIndexSet<NodeIndex>>),
+}
+
+fn analyze_commutations_inner(
+    py: Python,
+    dag: &mut DAGCircuit,
+    commutation_checker: &mut CommutationChecker,
+) -> HashMap<(Option<NodeIndex>, Wire), CommutationSetEntry> {
+    let mut commutation_set: HashMap<(Option<NodeIndex>, Wire), CommutationSetEntry> =
+        HashMap::new();
+    let max_num_qubits = 3;
+
+    (0..dag.num_qubits()).for_each(|qubit| {
+        let wire = Wire::Qubit(Qubit(qubit as u32));
+        dag.nodes_on_wire(py, &wire, false)
+            .iter()
+            .for_each(|current_gate_idx| {
+                if let CommutationSetEntry::SetExists(ref mut commutation_entry) = commutation_set
+                    .entry((None, wire.clone()))
+                    .or_insert_with(|| {
+                        CommutationSetEntry::SetExists(vec![AIndexSet::from_iter([
+                            *current_gate_idx,
+                        ])])
+                    })
+                {
+                    let last = commutation_entry.last_mut().unwrap();
+
+                    if !last.contains(current_gate_idx) {
+                        if last.iter().all(|prev_gate_idx| {
+                            if let (
+                                NodeType::Operation(packed_inst0),
+                                NodeType::Operation(packed_inst1),
+                            ) = (&dag.dag[*current_gate_idx], &dag.dag[*prev_gate_idx])
+                            {
+                                let empty_params: Box<SmallVec<[Param; 3]>> = Box::new(smallvec![]);
+                                let op1 = packed_inst0.op.view();
+                                let op2 = packed_inst1.op.view();
+                                let params1 = match packed_inst0.params.as_ref() {
+                                    Some(params) => params,
+                                    None => &empty_params,
+                                };
+                                let params2 = match packed_inst1.params.as_ref() {
+                                    Some(params) => params,
+                                    None => &empty_params,
+                                };
```

Comment:
You can just use `params_view()` for this:

```suggestion
                                let op1 = packed_inst0.op.view();
                                let op2 = packed_inst1.op.view();
                                let params1 = packed_inst0.params_view();
                                let params2 = packed_inst1.params_view();
```

Reply:
ahh perfect -- thanks!